ddtrace/_trace/processor/__init__.py

```diff
@@ -143,28 +143,29 @@ class TraceSamplingProcessor(TraceProcessor):
     """
 
     _compute_stats_enabled = attr.ib(type=bool)
+    sampler = attr.ib()
 
     def process_trace(self, trace):
         # type: (List[Span]) -> Optional[List[Span]]
         if trace:
+            chunk_root = trace[0]
+            root_ctx = chunk_root._context
+            chunk_root.sampled = self.sampler.sample(trace)
+            _update_span_tags_from_context(chunk_root, root_ctx)
```

Comment:
We run ` _update_span_tags_from_context` here and in TraceTagsProcessor. I'm a bit unsure if it's necessary. I think it may only be needed here, since I believe this runs before TraceTagsProcessor, and we don't seem modify the context between now and then. 

I am however also wondering if we need to run `sample` before we update the span tags. Potential fixes for this:
1. We look at context when sampling to check everything set on it
2. We update the root span beforehand

Reply:
Actually it's probably ok to remove this behavior since we have the sampler now look inside the context object for tags and metrics when sampling, so it doesn't need to all be on the root span when we sample.